DOC: update 1.16.0 release notes (#83)
tylerjereddy/scipy: doc/source/release/1.16.0-notes.rst

``scipy.differentiate`` improvements
====================================

``scipy.interpolate`` improvements
==================================
- Improved the error estimate of `scipy.integrate.tanhsinh`

``scipy.interpolate`` improvements
==================================

> - Added batch support to `scipy.interpolate.make_smoothing_spline`

Author: @ev-br were there other splines that got batch support, or was there anything else you wanted to add to the release notes? Feel free to add specific wording as a suggestion here (that can just be committed), or if this is merged first, please add to scipy/scipy#22991.

Reviewer: Doesn't this duplicate the `scipy.interpolate` heading from just above? I think I'll probably just merge this PR in a minute after looking it over in any case though, and just carry that comment over.

  deprecated and will be removed four versions later. Similarly, after
  changing the underlying algorithm to recursion, the ``blocksize`` keyword
  argument has no effect and will be removed two versions later.

> - Added wrappers for ``?langb`` and ``?gbcon`` to `scipy.linalg.lapack`.

Author: @j-boway I'm not sure if this is all of them. Thought you might want to take a look at the release notes.

``scipy.signal`` improvements
=============================

> -

Reviewer: spurious?

Experimental support for array libraries other than NumPy has been added to
existing sub-packages in recent versions of SciPy. Please consider testing
these features by setting an environment variable ``SCIPY_ARRAY_API=1`` and
providing PyTorch, JAX, or CuPy arrays as array arguments. Many functions

> in `scipy.stats`, `scipy.special`, `scipy.optimize`, and `scipy.constants`

Reviewer 2: ```suggestion
in `scipy.stats`, `scipy.special`, `scipy.optimize`, `scipy.constants`, and `scipy.cluster`
```
depending if https://github.com/scipy/scipy/pull/22960 gets merged

Author: K. If it's not merged before this PR merges, please keep this in mind as something to add to scipy/scipy#22991 directly.

now provide tables documenting compatible array and device types as well as

> support for lazy arrays and JIT compilation. New features with support and

Author: @crusaderky thought you might have something to add here. Please add suggestion that can be committed or feel free to follow up in scipy/scipy#22991.

> now provide tables documenting compatible array and device types as well as
> support for lazy arrays and JIT compilation. New features with support and

Reviewer 2: ```suggestion
now provide tables documenting compatible array and device types.
Many functions have gained support for JIT compilation in JAX and no longer
trigger an early computation in Dask, as documented by the above mentioned tables.
New features with support and
```

> old features with support added for SciPy 1.16.0 include:

Reviewer 2: ```suggestion
old features with support added or enhanced for SciPy 1.16.0 include:
```

Author: Feel free to make a separate PR with all enhancements to array API support. This PR only intends to list things that got some sort of support for the first time; otherwise, the list would be very long.

> - Most features of `scipy.signal`

Reviewer 2: ```suggestion
- Most features of `scipy.cluster`
- Most features of `scipy.signal`
```

Reviewer 2: Note: this is regardless of https://github.com/scipy/scipy/pull/22960

Author: Those were added before this release, though.

Reviewer 2: None of `cluster` worked with `jax.jit` in 1.15. Dask was not even a thing.

I'm changing `old features with support added` to `old features with support added or enhanced`.

Author: I didn't include new backends here. Then, as you say, essentially everything we have had in the past would appear on this list again. I'm just including functions that are getting translated for the first time.

- `scipy.ndimage.vectorized_filter`

> - `scipy.special.stdtrit`

Reviewer 2: ```suggestion
- `scipy.special.logsumexp`
- `scipy.special.softmax`
- `scipy.special.stdtrit`
```

Author: Also added before this relase, right?

Reviewer 2: Neither worked in jax.jit or Dask.

Reviewer 2: For the record: as much as end users are concerned, "worked in jax.jit" and "worked with JAX" should be treated almost as synonims. Not being able to apply the jit makes JAX very undesirable compared to numpy or cupy.

Author: Same as https://github.com/tylerjereddy/scipy/pull/83#discussion_r2093759143.
